_15thnight/core.py

```diff
@@ -17,26 +17,24 @@ def send_out_alert(alert_form):
     """
     alert = Alert(
         description=alert_form.description.data,
-        other=alert_form.other.data,
-        shelter=alert_form.shelter.data,
-        food=alert_form.food.data,
-        clothes=alert_form.clothes.data,
         gender=alert_form.gender.data,
         age=alert_form.age.data,
-        user=current_user
+        user=current_user,
+        categories=Category.get_by_ids(alert_form.categories.data).all()
     )
     alert.save()
-    providers = User.get_providers(
-        alert.food, alert.clothes, alert.shelter, alert.other
-    )
+    providers = User.query.join(User.categories) \
+        .filter(Category.id.in_(alert_form.categories.data)) \
+        .distinct()
```

Comment:
This is gross, should be condensed onto the `User` model as a simple method that takes the `alert_form.categories.data`.